tests/unit/test-util/test.manifest.js

```diff
@@ -158,6 +158,21 @@ describe('util/manifest', () => {
       }
     ));
 
+    it('allows comments in manifest JSON', () => withTempDir(
+      (tmpDir) => {
```

Comment:
We should also add an additional test that asserts that an error raised in a manifest with comments is still reported on the expected line number, we can probably use an assertion similar to the one used in the other test above:

```
assert.include(error.message, 'Unexpected token  in JSON at position 49');
```

and use a manifest with a comment that would potentially change the line, e.g. using something like:

```js
const invalidManifestWithComments = `{
  // a comment in its own line
// another other comment on its own line
  "name": "the extension
}`;
```

Besides that, do you mind to rewrite these new test cases using an async function? it would make these new tests much more readable (don't worry about the other test cases in this file, there's no need to rewrite all of them in this pull request, but it would be nice to don't add more of them).

it should be something like:

```js
it('allows comments in manifest JSON', () => 
  withTempDir(async (tmpDir) => {
    ...
    await fs.writefile(manifestfile, manifestWithComments);
    const manifestData = await getValidatedManifest(tmpDir.path());
    ...
  }));
```

Reply:
Fixed.  Your review comment is quite descriptive, so it's easy to understand for beginners in this area like me. :)
Thank you!